src/test/java/juja/microservices/users/dao/UserRepositoryTest.java

```diff
@@ -45,48 +47,71 @@
 
-        //then
-        assertEquals(2, users.size());
-        assertEquals("Alex", users.get(0).getFirstName());
-        assertEquals("Max", users.get(1).getFirstName());
+        assertEquals(1, users.size());
+        assertThat(users, contains(user1));
     }
 
     @Test
-    public void testFindBySlack() throws Exception {
-        //when
-        User user = repository.findOneBySlack("alex.batman");
+    public void testFindTwoUsersByThreeeSlackName() throws Exception {
+        List<User> expected = Arrays.asList(user1, user2);
+        List<String> slackNames = Arrays.asList(user1.getSlack(), user2.getSlack(), "fake.user");
 
-        //then
-        assertEquals("Batman Alex", user.getFullName());
-        assertEquals("alex.batman", user.getSlack());
+        List<User> actual = repository.findBySlackIn(slackNames);
+
+        assertThat(expected, is(actual));
+    }
+
+    @Test
+    public void testFindOneUserByUuid() throws Exception {
+        List<User> users = repository.findByUuidIn(Collections.singletonList(user1.getUuid()));
+
+        assertEquals(1, users.size());
+        assertThat(users, contains(user1));
     }
 
     @Test
-    public void testFindByUuid() throws Exception {
-        //when
+    public void testFindTwoUsersByUUID() throws Exception {
+        List<User> expected = Arrays.asList(user1, user2);
+        List<UUID> uuids = Arrays.asList(uuid1, uuid2);
+
+        List<User> actual = repository.findByUuidIn(uuids);
 
-        User user = repository.findOneByUuid(new UUID(1L, 3L));
-        //then
-        assertEquals("Superman Max", user.getFullName());
+        assertThat(expected, is(actual));
     }
 
     @Test
     @ExpectedDatabase(value = "/datasets/usersDataAfterUpdate.xml")
     public void testUpdateUsersDatabaseFromCRM() throws Exception {
-        //given
-        List<User> users = new ArrayList<>();
-        users.add(new User(UUID.fromString("00000000-0000-0001-0000-000000000003"), "Max", "Ironman",
-                "max.ironman", "Max", 200L));
-        users.add(new User(UUID.fromString("00000000-0000-0001-0000-000000000004"), "Sergey", "Spiderman",
-                "sergey.spiderman", "Sergey", 250L));
-
-        //when
-        repository.save(users);
+        User user1Updated = new User(UUID.fromString("00000000-0000-0001-0000-000000000003"), "Max", "Ironman",
+                "max.ironman", "Max", 200L);
+        User user4 = new User(UUID.fromString("00000000-0000-0001-0000-000000000004"), "Sergey", "Spiderman",
+                "sergey.spiderman", "Sergey", 250L);
+        List<User> expected = Arrays.asList(user1, user1Updated, user4);
+        List<User> newUsers = Arrays.asList(user1Updated, user4);
+
+        repository.save(newUsers);
         repository.flush();
+
+        List<User> actual = repository.findAll();
+        assertEquals(3, actual.size());
+        assertThat(actual, hasItems(expected.toArray(new User[expected.size()])));
```

Comment:
I agree with Vadim,
Why do you check the status of db again using the repository ?
for this goal we use dbunit and annotation @ExpectedDatabase(value = "/datasets/usersDataAfterUpdate.xml")

Reply:
I fix it